### PR TITLE
[V3] Plugins implemenations

### PR DESCRIPTION
**Pull request by atterpac**
Adds plugins as a service as discussed in #3496 
See discussion for more information until PR is finalized

### CODE REVIEW

File: v3/internal/assetserver/assetserver.go [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-	"path"
 	"strings"
 	"time"
 )
 
 const (
 	webViewRequestHeaderWindowId   = "x-wails-window-id"
 	webViewRequestHeaderWindowName = "x-wails-window-name"
-	pluginPrefix                   = "/wails/plugin"
+	servicePrefix                   = "/services"
```
Comment (leaanthony) [inline v3/internal/assetserver/assetserver.go:16]: I think we should keep "reserved" paths to a `/wails/` prefix. Thoughts?

Comment (atterpac) [reply v3/internal/assetserver/assetserver.go:16]: I have no issue with that

Comment (atterpac) [reply v3/internal/assetserver/assetserver.go]: Resolved in latest commit

**Comment by leaanthony** [comment]
@atterpac - any chance we could send those errors via the new error handlers? 🙏

**Comment by atterpac** [comment]
> @atterpac - any chance we could send those errors via the new error handlers? 🙏

Do we want to consider a plugin error as fatal?

**Review by leaanthony — APPROVED** [review]

**Pull request merged**